File: app/src/main/java/com/example/brijeshkum/mybaseproject/NetManager.java
package com.example.brijeshkum.mybaseproject;

import android.app.Application;
import android.content.Context;
import android.net.ConnectivityManager;

import javax.inject.Inject;

public final class NetManager {
    private Context context;

    public NetManager(Context context) {
        this.context = context;
    }

    public boolean isConnected(){
        ConnectivityManager cm = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        return cm.getActiveNetworkInfo().isConnectedOrConnecting();
    }
}

File: app/src/main/java/com/example/brijeshkum/mybaseproject/NetManager.kt
package com.example.brijeshkum.mybaseproject

import android.content.Context
import android.net.ConnectivityManager

class NetManager(private val context: Context) {
    val isConnected: Boolean
        get() {
            val cm = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
            return cm.activeNetworkInfo.isConnectedOrConnecting
        }

}